### DISCUSSION TITLE
Histogram Plot

### DISCUSSION THREAD

**Discussion by pietz** [f5/unovis#166] [Q&A] [answered]
Hi there,

is it currently possible to create a histogram plot, where a bar represents a range of values?

For reference: https://plotly.com/python/histograms/

**Comment by rokotyan** [comment, marked as answer]
@pietz You can create histograms using the [Stacked Bar](https://unovis.dev/docs/xy-charts/StackedBar) component; or using [Area](https://unovis.dev/docs/xy-charts/Area) with the [`curveType`](https://unovis.dev/docs/xy-charts/Area#curve-types) property set to `'step'`:
<img width="909" alt="image" src="https://user-images.githubusercontent.com/755708/227635577-a0ca6db9-3ae1-494e-a9bb-5b2c06471fe3.png">

You'll need to bin your data yourself though.

**Comment by pietz** [comment]
sorry to come back to this. i managed to do this with the area plot but would like to use the bar plot because i want there to be padding between the bars. how would you create a minimal example for plotting 3 values between 0 to 10 and 5 values between 11 and 20? basically i would want the ticks to be 0, 10 and 20 with 2 bars between the ticks. i can only manage to line up the ticks with the middle of the bars but not in between.

**Reply by rokotyan** [reply]
@pietz You can set tick values explicitly using the `tickValues` property of `Axis`. Here's the section in the docs about it: https://unovis.dev/docs/auxiliary/Axis/#set-ticks-explicitly. 
<img width="894" alt="image" src="https://user-images.githubusercontent.com/755708/231203043-4b9254fb-58c5-40cc-81a2-565143f0938c.png">


I believe that in your case `tickValues` should be set to `[0, 10, 20].`

**Reply by pietz** [reply]
I think what I'm looking for is more like the stepAfter and stepBefore options but using a bar chart.

**Reply by rokotyan** [reply]
@pietz I see, there's no option like that unfortunately, the bars will always be centered. But you can try manually shifting them by the half of your step size. It can be achieved by modifying the `x` accessor function like this: `x={(d) => d.x + 5}`

**Reply by pietz** [reply]
dont you think what I'm suggesting might be a good addition to unovis?

**Reply by rokotyan** [reply]
I think it can be a nice addition. We're open to contributions on that front if you want to help building it.

**Reply by pietz** [reply]
actually i took your advice and got everything working with the shifting method! it felt a bit hacky but it does exactly what i expect. very impressed with unovis! thanks for building such a great library :)
